src/main/java/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,21 @@
+package strategy;
+
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> inventory, String fruit, int quantity) {
+        int currentQuantity = inventory.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Not enough " + fruit + " in inventory.");
+        }
+
+        int newQuantity = currentQuantity - quantity;
+
+        if (newQuantity < 0) {
```

Comment:
This check is redundant because you have already checked if the currentQuantity is less than the quantity on lines 9-11. If the code reaches line 13, newQuantity will never be less than 0, so you can safely remove lines 15-17.